src/test/java/com/venafi/vcert/sdk/connectors/tpp/TppConnectorIT.java

```diff
@@ -0,0 +1,88 @@
+package com.venafi.vcert.sdk.connectors.tpp;
+
+import com.github.jenspiegsa.wiremockextension.InjectServer;
+import com.github.jenspiegsa.wiremockextension.WireMockExtension;
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.venafi.vcert.sdk.VCertException;
+import com.venafi.vcert.sdk.endpoint.Authentication;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import static com.venafi.vcert.sdk.SignatureAlgorithm.MD2withRSA;
+import static com.venafi.vcert.sdk.certificate.EllipticCurve.*;
+import static com.venafi.vcert.sdk.certificate.KeyType.ECDSA;
+import static com.venafi.vcert.sdk.certificate.KeyType.RSA;
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+@ExtendWith(WireMockExtension.class)
+class TppConnectorIT {
+
+    @InjectServer
+    private WireMockServer serverMock;
+
+    private TppConnector classUnderTest;
+
+    @BeforeEach
+    void setup() throws VCertException {
+        classUnderTest =  new TppConnector(Tpp.connect("http://localhost:" + serverMock.port() + "/vedsdk/")); // todo String.format()
+        Authentication auth = new Authentication("user", "pass", null);
+        classUnderTest.authenticate(auth);
+    }
+
+    @Test
+    @DisplayName("should start and inject server.")
+    void shouldInjectServer() {
+        assertThat(serverMock).isNotNull();
+        assertThat(serverMock.isRunning()).describedAs("server expected to be running.").isTrue();
+    }
+
+    @Test
+    void authenticate() throws VCertException {
+        // call in @BeforeEach
+        assertThat(classUnderTest.bestBeforeEnd).isEqualTo("2019-03-27T19:06:19.479Z");
+        assertThat(classUnderTest.apiKey()).isEqualTo("12345678-1234-1234-1234-123456789012");
+    }
+
+    @Test
+    void readZoneConfiguration() throws VCertException {
+        ZoneConfiguration zoneConfiguration = classUnderTest.readZoneConfiguration("tag");
+
+        assertThat(zoneConfiguration).isNotNull();
+        assertThat(zoneConfiguration.organization()).isNull();
+        assertThat(zoneConfiguration.organizationalUnit()).isNotNull();
+        assertThat(zoneConfiguration.organizationalUnit()).isEmpty();
+        assertThat(zoneConfiguration.country()).isNull();
+        assertThat(zoneConfiguration.province()).isNull();
+        assertThat(zoneConfiguration.locality()).isNull();
+        assertThat(zoneConfiguration.policy()).isNotNull();
+        assertThat(zoneConfiguration.policy().subjectCNRegexes()).containsExactly(".*");
```

Comment:
i don't know how and what type of regular expressions dialect do you use, but remember, that regexp should match full string. so for example regexp `.*\.example\.com` should doesn't match cn `www.example.company`

Reply:
Hello I see, in this case, the test is validating that the API requires its correct from a stub external service https://github.com/opencredo/vcert-java-1/blob/68f1d07aaf4fc61e613361f68fb43b5932712cea/src/test/resources/mappings/cloud.certificatepolicies.di.json#L40

Reply:
I don`t big specialist in java. I did`t know what java `pattern.matches` which match whole string. In golang you need to add `^` and `$` to your regexp for matching whole string.